tools/pathwaymatcher/pathwaymatcher.xml

```diff
@@ -0,0 +1,381 @@
+                          It is composed by the protein UniProt accession, isoform and set of post translational modifications.
+                          The input file contains one line for each proteoform. Each PTM is specified using a modification
+                          identifier and a site, separated by ':'(semicolon). For example: '00046:133'.
+                          The identifier is a 5 digit id from the PSI-MOD Protein Modification Onthology [6]."/>
+
+                      <param name="proteoform_match_criteria" type="select" label="Proteoform match criteria">
+                          <option value="STRICT">STRICT</option>
+                          <option value="ONE">ONE</option>
+                          <option value="SUPERSET" selected="True">SUPERSET</option>
+                          <option value="SUBSET">SUBSET</option>
+                      </param>
+                  </when>
+
+            </conditional>
+
+        </repeat>
+
+        <section name="output_options" expanded="true" title="Output options">
+
+            <param name="search_top_level_info" type="select" label="Add search top level info">
+                <option value="0" selected="True">False</option>
+                <option value="1">True</option>
+            </param>
+
+            <param name="output_graphs" type="select" display="checkboxes" multiple="True" label="Connection graphs"
+               help="Generates a zipped file with connection graphs as an additional output when executing the pathway search and analysis.
+               The graph can use genes, proteins or proteoforms as vertices.">
+                <option value="gg">Genes</option>
+                <option value="gu">Proteins</option>
+                <option value="gp">Proteoforms</option>
+            </param>
+
+        </section>
+
+    </inputs>
+    <outputs>
+        <data name="search" format="tsv" from_work_dir="search.tsv" label="${tool.name} - search on ${on_string}" />
+        <data name="analysis" format="tsv" from_work_dir="analysis.tsv" label="${tool.name} - analysis on ${on_string}" />
+        <collection name="graphs_files" type="list" label="${tool.name} - graphs on ${on_string}" >
+            <filter>output_options['output_graphs'] != None</filter>
+            <discover_datasets pattern="__name_and_ext__" directory="graphs" ext="tsv"/>
+        </collection>
+    </outputs>
+
+
+    <tests>
+
+      <!-- Test that genes search works -->
+      <test>
+          <repeat name="input_types">
+              <conditional name="input_type">
+                  <param name="input_type_selector" value="gene"/>
+                  <param name="input_gene" value="genes.txt" ftype="txt" />
+              </conditional>
+          </repeat>
+          <output name="search" file="genes_search.tsv" ftype="tsv" compare="sim_size" delta="3000" />
+      </test>
+
+      <!-- Test graphs from proteoforms -->
+      <test>
```

Comment:
You could add `<test expect_num_outputs="3">` to indicate the number of outputs and test for these.

Reply:
Um...it`s weird, I`ve done it (expect_num_outputs="3" in test 2) but it fails:

Exception: Incorrect number of outputs - expected 3, found 2.

expect_num_outputs="2" in test 1 works.
Maybe doesn`t collections count for this condition (expect_num_outputs)?

Reply:
mh good question, then leave it out for the colleciton stuff.